contracts/LRTDepositPool.sol

```diff
@@ -164,13 +174,96 @@ contract LRTDepositPool is ILRTDepositPool, LRTConfigRoleChecker, PausableUpgrad
     }
 
     /// @dev private function to mint primeETH
-    /// @param primeEthAmount Amount of primeETH minted
-    function _mint(uint256 primeEthAmount) private {
+    /// @param amount Amount of primeETH to be minted
+    function _mint(uint256 amount) private {
         address primeETH = lrtConfig.primeETH();
         // mint primeETH for user
-        IPrimeETH(primeETH).mint(msg.sender, primeEthAmount);
+        IPrimeETH(primeETH).mint(msg.sender, amount);
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                        Withdraw functions
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice PrimeETH staker requests the withdrawal of a liquid staking token (LST) from
+    // its respective underlying EigenLayer strategy.
+    /// @dev Will emit the `Withdrawal` event from EigenLayer's `DelegationManager` contract
+    /// which is needed for the `claimWithdrawal` call.
+    /// @param asset address of the liquid staking token (LST) being requested. eg OETH. Can not be WETH.
+    /// @param assetAmount the amount of LSTs to withdraw.
+    /// @param maxPrimeETH the maximum amount of primeETH tokens that can be burned.
+    /// @return primeETHAmount the amount of primeETH tokens that were burned.
+    function requestWithdrawal(
+        address asset,
+        uint256 assetAmount,
+        uint256 maxPrimeETH
+    )
+        external
+        whenNotPaused
+        nonReentrant
+        onlySupportedAsset(asset)
+        returns (uint256 primeETHAmount)
+    {
+        if (assetAmount == 0) {
+            revert ZeroAmount();
+        }
+
+        // Only LST withdraws are currently supported
+        if (asset == WETH_TOKEN_ADDRESS) {
+            revert OnlyLSTWithdrawals();
+        }
+
+        // Convert asset amount to primeETH amount.
+        // Using mint function here even though it's a withdrawal as it does the same calculation
+        primeETHAmount = getMintAmount(asset, assetAmount);
+
+        // Check the primeETH amount to be burned is within the maximum allowed
+        if (primeETHAmount > maxPrimeETH) {
+            revert MaxBurnAmount();
+        }
+
+        // burn primeETH from the staker
+        address primeETH = lrtConfig.primeETH();
+        IPrimeETH(primeETH).burnFrom(msg.sender, primeETHAmount);
+
+        // Get the NodeDelegator contract to request the withdrawal from
+        address nodeDelegator = nodeDelegatorQueue[LST_NDC_INDEX];
+
+        // Calculate the strategy shares for the requested assets
+        address strategyAddress = lrtConfig.assetStrategy(asset);
```

Comment:
Is there a need for a check to make sure the strategy holding the asset has enough quantity of it to proceed with the withdrawal?

Reply:
@naddison36 Can you confirm that requesting a withdraw that is too large will result in a revert from eigenlayer?

Reply:
Most of the fork tests are in the Holesky tests as Mainnet hasn't been upgraded to EigenLayer's M2 contracts at the time of development.

`test_revertRequestWithdrawalStrategyLiquidity()` covers not enough LST in EigenLayer.
EigenLayer reverts with `StrategyManager._removeShares: shareAmount too high`

Other requestWithdrawal tests against Holesky are:
// requestWithdrawal when LRTDepositPool paused
// requestWithdrawal when NodeDelegator paused
// requestWithdrawal with not enough primeETH
// requestWithdrawal with zero amount
// requestWithdrawal over max primeETH amount
// requestWithdrawal with WETH
// requestWithdrawal with unsupported LST asset
// requestWithdrawal with not enough LST in EigenLayer
// malicious user calls requestWithdrawal on the NodeDelegator